### PR TITLE
Logic and Name Changes

### PR DESCRIPTION
**Pull request by Vodzilla**
Fixes several logic bugs and changes names of several locations to make them more user friendly for merging into main branch.

### CODE REVIEW

File: data/World/Bottom of the Well.json [hunk 1]
```diff
@@ -42,8 +42,8 @@
             "Bottom of the Well Left Side Pot 3": "True",
             "Bottom of the Well Near Entrance Pot 1": "True",
             "Bottom of the Well Near Entrance Pot 2": "True",
-            "Bottom of the Well Fire Keese Pot" : "(Small_Key_Bottom_of_the_Well, 3) and (logic_lens_botw or can_use(Lens_of_Truth))",
```
Comment (rrealmuto) [inline data/World/Bottom of the Well.json:45]: Should we not have the lens logic here?

File: data/World/Dodongos Cavern.json [hunk 2]
```diff
@@ -31,10 +31,18 @@
             "Dodongos Cavern Right Side Pot 4": "here(can_blast_or_smash or Progressive_Strength_Upgrade) or is_adult",
             "Dodongos Cavern Right Side Pot 5": "here(can_blast_or_smash or Progressive_Strength_Upgrade) or is_adult",
             "Dodongos Cavern Right Side Pot 6": "here(can_blast_or_smash or Progressive_Strength_Upgrade) or is_adult",
-            "Dodongos Cavern Torch Room Pot 1": "((here(can_blast_or_smash) or Progressive_Strength_Upgrade) and can_child_attack) or is_adult",
-            "Dodongos Cavern Torch Room Pot 2": "((here(can_blast_or_smash) or Progressive_Strength_Upgrade) and can_child_attack) or is_adult",
-            "Dodongos Cavern Torch Room Pot 3": "((here(can_blast_or_smash) or Progressive_Strength_Upgrade) and can_child_attack) or is_adult",
-            "Dodongos Cavern Torch Room Pot 4": "((here(can_blast_or_smash) or Progressive_Strength_Upgrade) and can_child_attack) or is_adult",
+            "Dodongos Cavern Torch Room Pot 1": "
+                (has_explosives or Progressive_Strength_Upgrade) and
```
Comment (rrealmuto) [inline data/World/Dodongos Cavern.json]: What's the reasoning for removing can_child_attack and replacing w/ the individual items? Does Dins Fire not hurt the lizalfos?

File: data/World/Ice Cavern.json [hunk 3]
```diff
@@ -28,20 +28,18 @@
             "Ice Cavern Map Room Recovery Heart 1": "is_adult",
             "Ice Cavern Map Room Recovery Heart 2": "is_adult",
             "Ice Cavern Map Room Recovery Heart 3": "is_adult",
-            "Ice Cavern Block Room Red Rupee 1": "(is_adult and can_play(Song_of_Time)) or (is_child and can_use(Boomerang))",
-            "Ice Cavern Block Room Red Rupee 2": "(is_adult and can_play(Song_of_Time)) or (is_child and can_use(Boomerang))",
-            "Ice Cavern Block Room Red Rupee 3": "(is_adult and can_play(Song_of_Time)) or (is_child and can_use(Boomerang))",
+            "Ice Cavern Block Room Red Rupee 1": "Blue_Fire and (is_adult and can_play(Song_of_Time)) or (is_child and can_use(Boomerang))",
```
Comment (rrealmuto) [inline data/World/Ice Cavern.json]: Need additional parentheses

File: data/World/Dodongos Cavern.json (hunk 2 above)
Comment (rrealmuto) [reply data/World/Dodongos Cavern.json]: Also, while not strictly necessary, the entire "and" statement should be in parentheses.

`                ((has_explosives or Progressive_Strength_Upgrade) and
                (Slingshot or Sticks or Bombs or Kokiri_Sword)) or is_adult",`

**Review by rrealmuto — COMMENTED** [review]
See comments

File: data/World/Ice Cavern.json (hunk 3 above)
Comment (rrealmuto) [reply data/World/Ice Cavern.json]: `"Blue_Fire and ((is_adult and can_play(Song_of_Time)) or (is_child and can_use(Boomerang)))"`

Comment (rrealmuto) [reply data/World/Ice Cavern.json]: also while we're reviewing this one. Can child get these with SoT?

File: data/World/Bottom of the Well.json (hunk 1 above)
Comment (Vodzilla) [reply data/World/Bottom of the Well.json:45]: I don't think it's required.  There's a pretty generous strip of solid ground running from the pot when you hug the wall.  Lens logic can be re-added if you think it's necessary.

File: data/World/Dodongos Cavern.json (hunk 2 above)
Comment (Vodzilla) [reply data/World/Dodongos Cavern.json]: I followed the existing logic for getting to the scrub, which had all those individual items.  I can add the parentheses.

File: data/World/Bottom of the Well.json (hunk 1 above)
Comment (rrealmuto) [reply data/World/Bottom of the Well.json:45]: i think it should probably be kept in. I assume most people play w/ the logic setting anyway?

File: data/World/Ice Cavern.json (hunk 3 above)
Comment (Vodzilla) [reply data/World/Ice Cavern.json]: Child can in fact get those with Song of Time.  Will add that and the parentheses

File: data/World/Bottom of the Well.json (hunk 1 above)
Comment (Vodzilla) [reply data/World/Bottom of the Well.json:45]: Re-added lens logic to the pot.

**Pull request merged**